packages/lesmis/src/api/queries/useVitalsData.js

```diff
@@ -8,100 +8,210 @@ import { get, post } from '../api';
 import { useEntityData } from './useEntityData';
 import { useEntitiesData } from './useEntitiesData';
 
-const getDecendantsOfType = (entities, rootEntityCode, type) => {
+const getParentOfType = (entities, rootEntityCode, type) => {
   const entity = entities.find(e => e.code === rootEntityCode);
-
+  if (!entity) {
+    return undefined;
+  }
+  if (entity.type === type) {
+    return entity;
+  }
+  if (!entity.parentCode) {
+    return undefined;
+  }
+  return getParentOfType(entities, entity.parentCode, type);
+};
+const getDecendantCodesOfType = (entities, rootEntityCode, type) => {
+  const entity = entities?.find(e => e.code === rootEntityCode);
   if (!entity) {
     return [];
   }
-
   if (entity.type === type) {
-    return [entity.code];
+    return [entity?.code];
   }
-
   if (!entity.childCodes) {
     return [];
   }
-  return entity.childCodes.map(c => getDecendantsOfType(entities, c, type)).flat();
+  return entity.childCodes.map(c => getDecendantCodesOfType(entities, c, type)).flat();
 };
 
-const useSchoolInformation = entityCode =>
+const useReport = (entity, reportName, options, enabled) =>
   useQuery(
-    ['vitals', 'school', entityCode],
-    () =>
-      get(`reportData/${entityCode}/LESMIS_school_vitals`, {
-        params: { endDate: '2021-01-01' },
-      }),
+    [reportName, entity?.code],
```

Comment:
```suggestion
    [reportName, entity?.code, options],
```

Reply:
Need to include the options, otherwise we will get cached results even when the options are different.

Reply:
Also needed to swap out the formatting of `endTime` just because I had been passing it as the current `Date()` which would include the current time and therefore never actually return a cached value if we passed it as the option.